src/FlatButton/FlatButton.js

```diff
@@ -221,7 +221,7 @@ class FlatButton extends Component {
 
     if (icon) {
       iconCloned = React.cloneElement(icon, {
-        color: mergedRootStyles.color,
+        color: icon.props.color,
```

Comment:
Oh yeah..shoot. I missed the case when there is not color prop passed. 

Thanks @mbrookes for pointing it out.